sktime/forecasting/regularized_var.py

```diff
@@ -0,0 +1,224 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements Regularized (L1/L2) VAR using cvxpy."""
+
+# __author__ = [meraldoantonio]
+
+import numpy as np
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class RegularizedVAR(BaseForecaster):
+    """Custom forecaster using VAR with regularization.
+
+    Parameters
+    ----------
+    lags : int
+        The number of lagged values to include in the model.
+    L1_penalty : float, optional (default=0.0)
+        The L1 regularization penalty.
+    L2_penalty : float, optional (default=0.0)
+        The L2 regularization penalty.
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["meraldoantonio"],
+        "python_dependencies": "cvxpy",
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+    }
+
+    def __init__(self, lags, L1_penalty=0.0, L2_penalty=0.0):
+        self.lags = lags
+        self.L1_penalty = L1_penalty
+        self.L2_penalty = L2_penalty
+        self.coefficients = None
+        self.intercept = None
+        self.num_series = None
+        super().__init__()
+
+    def create_var_data(self, data):
+        """
+        Prepare the data for VAR(p) model.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            The input time series data.
+
+        Returns
+        -------
+        X : np.ndarray
```

Comment:
I think exact sizes and a description of what this does would be very helpful for a reviewer

Reply:
Noted, added shapes of the returned array and description of the function